### ISSUE TITLE
Singularity always display a wrong warning when seccomp is disabled

### ISSUE THREAD

**Issue by cclerget** [apptainer/singularity#3418]
## Version of Singularity:

3.2.0-rc2

### Expected behavior

No warning

### Actual behavior

Always display seccomp even if not requested with `--security=seccomp`

`WARNING: seccomp requested but not enabled, seccomp library is missing or too old`

### Steps to reproduce behavior

Compile singularity without seccomp library/headers

**Comment by 0xaf1f** [comment]
I'm seeing this issue in the official release 3.2.0-1 on CentOS 7.

**Comment by A33a** [comment]
I am also seeing this on Scientific Linux 7.3 using the release 3.2.0-1.

```bash
hostname:123 11:23:50 ~ > singularity exec sing.sif bash
WARNING: seccomp requested but not enabled, seccomp library is missing or too old
```

**Comment by Amir-Arsalan** [comment]
@GodloveD I am also getting the same error message. Should I set a flag to enable seccomp when building my container?

**Comment by truatpasteurdotfr** [comment]
I see that on our cluster campus running centos6:
```
[tru@tars-submit0 ~]$ singularity --version
singularity version 3.2.0-1
[tru@tars-submit0 ~]$ singularity shell docker://alpine
INFO:    Converting OCI blobs to SIF format
INFO:    Starting build...
Getting image source signatures
Copying blob sha256:e7c96db7181be991f19a9fb6975cdbbd73c65f4a2681348e63a141a2192a5f10
 2.63 MiB / 2.63 MiB [======================================================] 0s
Copying config sha256:8c79fc709348f9fdb30cc2dc10999b30e095fc7cad8aa9320e8834aca05f7740
 585 B / 585 B [============================================================] 0s
Writing manifest to image destination
Storing signatures
INFO:    Creating SIF file...
INFO:    Build complete: /pasteur/homes/tru/.singularity/cache/oci-tmp/769fddc7cc2f0a1c35abb2f91432e8beecf83916c421420e6a6da9f8975464b6/alpine_latest.sif
WARNING: seccomp requested but not enabled, seccomp library is missing or too old
Singularity>
```

I don't see it on my centos7 machines (same singularity version).

**Comment by DrDaveD** [comment]
I also see the warning on centos6 but not centos7.

**Comment by DrDaveD** [comment]
I believe the problem is that pr #3419 did not get cherry-picked into 3.2.0 despite the labels on this issue.  I plan to include it in the EPEL build.

**Comment by mcuma** [comment]
Can someone please tell us if/when this will be fixed in the 3.2.0 source? We are holding off releasing 3.2.0 to the users because of this (don't want to confuse them), and using patched 3.1.1 instead for now. 
We are seeing this in CentOS 7.6 despite installing libseccomp-devel and building 3.2.0 with the seccomp enabled.
Thanks.

**Comment by DrDaveD** [comment]
@mcuma 3.2.0 is what it is; it can't be changed once released.  You can try adding the #3419 PR patch to your build to see if that helps for you on centos7, but I don't understand the root cause there because I can't duplicate your experience.   If you want a src rpm with the patch, here's the [Fedora rawhide build](https://koji.fedoraproject.org/koji/taskinfo?taskID=34959048) that I just made.

**Comment by mcuma** [comment]
Thanks @DrDaveD, I was thinking of that and just did it, which has removed the warning. 

From what I could gather from the changes, config.Linux.Seccomp was not set earlier so seccomp.Enabled() was false, triggering the warning.

**Comment by cclerget** [comment]
Fixed by #3419
